docs/guides/error-handling.mdx

```diff
@@ -65,77 +69,255 @@ RESTATE_WORKER__INVOKER__RETRY_POLICY__INTERVAL=100ms \
+If the caller is another Restate service, the terminal error will propagate across RPCs, and will get thrown at the line where the RPC was made.
+If this is not caught, it will propagate further up the call stack until it reaches the original caller.
+
+You can catch terminal errors just like any other error, and build control flow around this.
+For example, the catch block can run undo actions for the actions you did earlier in your handler, to bring it to a consistent state before rethrowing the terminal error.
+
+For example, to catch a terminal error of a run block:
+<CodeWithTabs groupId={"sdk"}>
+    ```ts !!tabs TypeScript
+    CODE_LOAD::ts/src/guides/retries.ts#catch
+    ```
+
+    ```python !!tabs Python
+    CODE_LOAD::python/src/guides/retries.py#catch
+    ```
+
+    ```java !!tabs Java
+    CODE_LOAD::java/src/main/java/guides/RetryRunService.java#catch
+    ```
+
+    ```kotlin !!tabs Kotlin
+    CODE_LOAD::kotlin/src/main/kotlin/guides/RetryRunService.kt#catch
+    ```
+
+    ```go !!tabs Go
+    CODE_LOAD::go/guides/retries.go#catch
+    ```
+
+    ```rust !!tabs Rust
+    CODE_LOAD::rust/src/guides/retries.rs#catch
+    ```
+</CodeWithTabs>
+
+<Admonition type="note" title="Sagas with Restate">
+    When you throw a terminal error, you might need to undo the actions you did earlier in your handler to make sure that your system remains in a consistent state.
+    Have a look at our [sagas guide](/guides/sagas) to learn more.
+</Admonition>
+
+
+## Cancellations are Terminal Errors
+
+You can cancel invocations via the [CLI](/operate/invocation#cancelling-invocations), UI and programmatically.
+When you cancel an invocation, it throws a terminal error in the handler processing the invocation the next time it awaits a Promise or Future of a Restate Context action (e.g. run block, RPC, sleep,...; `CombineablePromise` in TypeScript, `Awaitable` in Java).
+Unless caught, This terminal error will propagate up the call stack until it reaches the original caller.
+
+Here again, the handler needs to have [compensation logic](/guides/sagas) in place to make sure the system remains in a consistent state, when you cancel an invocation.
 
 ## Timeouts to bound response times
 
+Restate also has timeouts to bound the response time of a service invocation.
+
+There are two types of timeouts in Restate:
+
 ### Inactivity timeout
-Default 1 minute
 
+When the Restate Server does not receive a next journal entry from a running handler within the inactivity timeout, it will ask the handler to suspend.
 This timer guards against stalled service/handler invocations. Once it expires, Restate triggers a graceful termination by asking the service invocation to suspend (which preserves intermediate progress).
 
-The 'abort timeout' is used to abort the invocation, in case it doesn't react to the request to suspend.
-
-Can be configured using the humantime format.
+By default, the inactivity timeout is set to one minute.
```

Comment:
I think the documentation should tell you why you want to increase it, and why you want to decrease it.

In the case of the inactivity timeout, I think it should be something like the following:

* Increase it when you can afford many idle HTTP requests, and you want to avoid too many replays, especially with invocations with long journals.
* Decrease it when you want to free up idle HTTP requests as much as possible, and you're ok with having too many replays, and your invocations generally progress fast and don't stay idle too long